src/lightning/fabric/connector.py

```diff
@@ -220,10 +226,23 @@ def _check_config_and_set_final_flags(
 
         self._accelerator_flag = accelerator
 
-        supported_precision = get_args(_PRECISION_INPUT_STR) + get_args(_PRECISION_INPUT_INT)
+        supported_precision = (
+            get_args(_PRECISION_INPUT_STR) + get_args(_PRECISION_INPUT_INT) + get_args(_PRECISION_INPUT_STR_LEGACY)
+        )
         if precision not in supported_precision:
             raise ValueError(f"Precision {repr(precision)} is invalid. Allowed precision values: {supported_precision}")
-        self._precision_input = cast(_PRECISION_INPUT_STR, str(precision))
+
+        precision = str(precision)  # convert int flags to str here to enable the legacy-conversion below
+
+        if precision in get_args(_PRECISION_INPUT_STR_LEGACY):
+            if str(precision)[:2] not in ("32", "64"):
```

Comment:
Also, shouldn't 32 and 64 be removed from `_PRECISION_INPUT_STR_LEGACY`? Then this `if` wouldn't be needed

Reply:
but then the conversion logic wouldn't work anymore. Currently the plugins only need to implement the `_PRECISION_INPUT_STR` values and everything else is handled by the connector.

if we removed it, the plugins would have to deal with duplicates.